apps/onnx/CMakeLists.txt

```diff
@@ -0,0 +1,162 @@
+cmake_minimum_required(VERSION 3.28)
+project(onnx)
+
+enable_testing()
+
+# Set up language settings
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED YES)
+set(CMAKE_CXX_EXTENSIONS NO)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+find_package(Protobuf)
+if (NOT Protobuf_FOUND)
+    message(WARNING "Could NOT find Protobuf")
+    return()
+endif ()
+
+find_package(Halide REQUIRED)
+
+# Download onnx.proto
+set(ONNX_PROTO_FILE ${CMAKE_CURRENT_BINARY_DIR}/onnx.proto)
+if (NOT EXISTS ${ONNX_PROTO_FILE})
+    file(
+        DOWNLOAD
+        "https://raw.githubusercontent.com/onnx/onnx/v1.4.1/onnx/onnx.proto"
+        ${ONNX_PROTO_FILE}
+        SHOW_PROGRESS
+        STATUS download_status
+    )
+    if (NOT download_status EQUAL 0)
+        message(FATAL_ERROR "Failed to download onnx.proto: ${download_status}")
+    endif ()
+
+    file(READ ${ONNX_PROTO_FILE} onnx_proto_content)
+    string(REPLACE "package onnx;" "package onnx;option optimize_for = LITE_RUNTIME;" onnx_proto_content "${onnx_proto_content}")
+    file(WRITE ${ONNX_PROTO_FILE} "${onnx_proto_content}")
+endif ()
+
+# Generate sources from onnx.proto
+add_custom_target(onnx_proto)
+protobuf_generate(
+    TARGET onnx_proto
+    LANGUAGE cpp
+    PROTOC_OUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/onnx
+    PROTOC_OUT_VAR onnx_proto_generated_sources
+    PROTOS ${ONNX_PROTO_FILE}
+    IMPORT_DIRS ${CMAKE_CURRENT_BINARY_DIR}
+)
+
+# Add library that converts ONNX models to Halide operators
+add_library(oclib STATIC onnx_converter.cc ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx.pb.cc)
+target_include_directories(
+    oclib
+    PUBLIC
+    ${CMAKE_CURRENT_BINARY_DIR}
+    ${Protobuf_INCLUDE_DIRS}
+)
```

Comment:
We can probably simplify all this and the custom target that drives protobuf by using `FILE_SETS`.